### PR TITLE
Docs: Make compatibility example consistent.

### PR DESCRIPTION
**Pull request by emkornfield**
* make implementation match definition in interface
* clarify why adding a new interface method breaks compatibility.

### CODE REVIEW

**Comment by emkornfield** [comment]
@Fokko would you mind taking a look?

File: site/docs/contribute.md [hunk 1]
```diff
@@ -196,6 +196,8 @@ public interface ManageSnapshots extends PendingUpdate<Snapshot> {
   // existing code...
 
   // adding this method introduces an API-breaking change
+  // since existing classes implementing ManagedSnapshots
```
Comment (amogh-jahagirdar) [inline site/docs/contribute.md:199]:
```suggestion
    // since existing classes implementing ManageSnapshots
```

Comment (amogh-jahagirdar) [inline site/docs/contribute.md:199]:
```suggestion
  // since existing classes implementing ManageSnapshots
```

**Review by amogh-jahagirdar — APPROVED** [review]
Thanks @emkornfield!

**Pull request merged**